pkg/cloudprovider/aws/block_store.go

```diff
@@ -180,6 +190,29 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 	return *res.SnapshotId, nil
 }
 
+func getTagsForCluster(clusterTags map[string]string, snapshotTags []*ec2.Tag) []*ec2.Tag {
+	var result []*ec2.Tag
+
+	// set current cluster ownership tags
+	for k, v := range clusterTags {
+		result = append(result, ec2Tag(k, v))
+	}
+
+	for _, tag := range snapshotTags {
+		if clusterNameTag, _ := regexp.MatchString("kubernetes.io/cluster/*", *tag.Key); clusterNameTag {
+			// if the AWS_CLUSTER_NAME variable is found we want current cluster
+			// to overwrite the old ownership on volumes
+			if _, found := os.LookupEnv("AWS_CLUSTER_NAME"); found {
```

Comment:
Do this check once, outside the loop

Reply:
Hi, thank you for the review and your suggestion under #451! By doing this check once you mean this:
`	if _, found := os.LookupEnv("AWS_CLUSTER_NAME"); found {
		for _, tag := range snapshotTags {
			if clusterNameTag, _ := regexp.MatchString("kubernetes.io/cluster/*", *tag.Key); clusterNameTag {
				// if the AWS_CLUSTER_NAME variable is found we want current cluster
				// to overwrite the old ownership on volumes
				continue
			}

			result = append(result, ec2Tag(*tag.Key, *tag.Value))
		}
	} else {
		for _, tag := range snapshotTags {
			result = append(result, ec2Tag(*tag.Key, *tag.Value))
		}
	}`

Reply:
More like this:

```go
        haveAWSClusterNameEnvVar := os.GetEnv("AWS_CLUSTER_NAME") != ""

        for _, tag := range snapshotTags {
 		if haveAWSClusterNameEnvVar && strings.HasPrefix(*tag.Key, "kubernetes.io/cluster/") {
 			// if the AWS_CLUSTER_NAME variable is found we want current cluster
 			// to overwrite the old ownership on volumes
 			continue
 		}

 		result = append(result, ec2Tag(*tag.Key, *tag.Value))
 	}
```

Reply:
Shame on me for not figuring this out, thank you very much! :)